### ISSUE TITLE
Internal territory estimator

### ISSUE THREAD

**Issue by TFiFiE** [gcp/leela-zero#1013]
To evaluate a board position the neural network must surely effectively make an assessment of how much each intersection is expected to belong to one or the other player. Is it reasonable to expect a simple linear combination of (intermediate) values to make up this territory estimator and could it be worth trying to isolate it?
Related to #98.

**Comment by gcp** [comment]
I would look at the output of the final convolution for the value head. It's only a single go board, so odds are...

**Comment by TFiFiE** [comment]
A cursory investigation suggests that that does actually somewhat match the actual territorial division, but it tends to get edges and borders wrong.

**Comment by TFiFiE** [comment]
Is there, perhaps for visualization purposes, some desire here for these pseudo-evaluations of individual intersections to be made available to users (possibly after some massaging), even if some of them are off?

**Comment by WYVYW** [comment]
@TFiFiE  Could you add that pseudo-evaluation to lizzie?
I would like to see what winrate translates to score even if some are off (also to see if and how it progresses).

**Comment by TFiFiE** [comment]
I'm really not that enamored with the accuracy of these pseudo-evaluations _as is_ and had been toying with the idea of territory estimation being dynamically evaluated as part of the search, especially in light of the fact the use of random symmetries might help smooth out the inaccuracies. For all I know, however, this could still be a case of garbage in, garbage out.

**Comment by TFiFiE** [comment]
Even when dynamically aggregated, this kind of territory estimation is too "smudged" for my taste to work satisfactorily. It's important to note that the policy and value share the same computation until right before the end and that could very well explain the distortion.

My advice for people would be to lobby for #98 to become part of the next architectural change, because I can't imagine a better way of making Leela Zero more useful as an analysis tool.

**Comment by Hersmunch** [comment]
I agree. I had a quick go at applying the weights for the value head on a per square basis to the value output data of the network. https://github.com/Hersmunch/leela-zero/commit/8d79c94f1fe4dca713cb42d7a5588ceb74ccc9d1
I mean I very naively equally divided the bias between all the squares. It didn't help much.

From what I could make out it looks like the data contains more than just a value based upon which side is likely winning each square. There are rows and column that look like they have an offset or something. I haven't tried it on many positions but on a won game, the winning side's values look like they are saturating toward that side's winning values. The losing side's output looks more reasonable. They both seem to count their own territory and regard dame and the opponents squares as the same (but you can just subtract one from the other because of the global offsetting of the winning side.
Does that sound like the same from what you saw @TFiFiE ?
I guess without something like #98 there is no reason for the network to try and make sense of the overall value on a square by square basis as it isn't a training optimisation goal.
(I'll try to take some screenshots when I have a sec)

**Comment by WYVYW** [comment]
One goal is to improve LZ. An estimation option (just as any other improvement which brings LZ closer to be a training tool) could lure more people to use LZ and support LZ with computing power. 
I am no coder, so I don't know how much work is involved, but if it is not a big deal to activate that option, I would implement it, even if it is not reliable.

**Comment by TFiFiE** [comment]
I'm going to have to disagree there, because if the estimation is clearly off to the point it's even obvious to weak players, then that will reflect badly on the engine, regardless of all disclaimers of its speculative nature.

**Comment by alreadydone** [comment]
https://www.reddit.com/r/cbaduk/comments/ad1w30/using_leela_zero_to_generate_influence/edfkup8/

**Comment by alreadydone** [comment]
Miraculously, deleting boundary lines from the fc layers in policy head (two lines with 361x2x362 and 362 numbers respectively) of LZ # 203 yields a 13x13 net with strong policy head (defeats [a trained 13x13 net](https://github.com/ihavnoid/leelaz-ninenine/issues/6#issuecomment-457889759) 6-1 under `-p 1` according to @hzyhhzy's implementation and tests). However doing the same to the value head fc layer (one line with 361x256 numbers) yields a broken value head. He also tried multiplying the conv weights (361x2x362) by 361/169 or other numbers but it didn't help much.
The above reddit post talks about lost of spatial information and checkboard pattern; that could be relevant and makes me wonder if some averaging is more appropriate. Since the policy head works well, the final residual tower output #filterx19x19 should be retaining some spatial information.
